dids/didweb/didweb.go

```diff
@@ -0,0 +1,157 @@
+		if o.services == nil {
+			o.services = make([]didcore.Service, 0)
+		}
+
+		o.services = append(o.services, svc)
+	}
+}
+
+// PrivateKey is used to add a private key to the DID being created with the [Create] function.
+// Each PrivateKey provided will be used to generate a private key in the key manager and then
+// added to the DID Document as a VerificationMethod.
+func PrivateKey(algorithmID string, purposes ...string) CreateOption {
+	return func(o *createOptions) {
+		keyOpts := privateKeyOption{algorithmID: algorithmID, purposes: purposes}
+
+		if o.privateKeys == nil {
+			o.privateKeys = make([]privateKeyOption, 0)
+		}
+
+		o.privateKeys = append(o.privateKeys, keyOpts)
+	}
+}
+
+// KeyManager is used to set the key manager that will be used to generate the private keys for the DID.
+func KeyManager(km crypto.KeyManager) CreateOption {
+	return func(o *createOptions) {
+		o.keyManager = km
+	}
+}
+
+// AlsoKnownAs is used to set the 'alsoKnownAs' property of the DID Document.
+// more details here: https://www.w3.org/TR/did-core/#also-known-as
+func AlsoKnownAs(aka ...string) CreateOption {
+	return func(o *createOptions) {
+		o.alsoKnownAs = aka
+	}
+}
+
+// Controllers is used to set the 'controller' property of the DID Document.
+// more details here: https://www.w3.org/TR/did-core/#controller
+func Controllers(controllers ...string) CreateOption {
+	return func(o *createOptions) {
+		o.controllers = controllers
+	}
+}
+
+// Create creates a new 'did:web' BearerDID with the given domain and options provided.
+// If no options are provided, a default key manager will be used to generate a single ED25519 key pair.
+// The resulting public key will be added to the DID Document as a VerificationMethod.
+func Create(domain string, opts ...CreateOption) (_did.BearerDID, error) {
+	options := &createOptions{
+		keyManager: crypto.NewLocalKeyManager(),
+		privateKeys: []privateKeyOption{
+			{
+				algorithmID: dsa.AlgorithmIDED25519,
+			},
+		},
+	}
+
+	for _, opt := range opts {
+		opt(options)
+	}
+
+	did, err := _did.Parse("did:web:" + domain)
+	if err != nil {
+		return _did.BearerDID{}, fmt.Errorf("invalid domain: %w", err)
+	}
+
+	document := didcore.Document{
+		ID: did.URI,
+	}
+
+	if len(options.alsoKnownAs) > 0 {
+		document.AlsoKnownAs = options.alsoKnownAs
+	}
+
+	if len(options.controllers) > 0 {
+		document.Controller = options.controllers
+	}
+
+	for idx, keyOpts := range options.privateKeys {
+		keyID, err := options.keyManager.GeneratePrivateKey(keyOpts.algorithmID)
+		if err != nil {
+			return _did.BearerDID{}, fmt.Errorf("failed to generate %s private key: %w", keyOpts.algorithmID, err)
+		}
+
+		publicKeyJWK, err := options.keyManager.GetPublicKey(keyID)
+		if err != nil {
+			return _did.BearerDID{}, fmt.Errorf("failed to get public key for private key %s: %w", keyID, err)
+		}
+
+		vm := didcore.VerificationMethod{
+			ID:           "#" + fmt.Sprint(idx),
```

Comment:
uber nit: totally fine, but I think it would be a bit more idiomatic to use `strconv.Itoa()`.

Reply:
Oo I didn't know that even existed. Thanks for the suggestion. Will update